src/component-library/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,71 @@
+import { useFocusRing } from '@react-aria/focus';
+import { useTabList } from '@react-aria/tabs';
+import { mergeProps } from '@react-aria/utils';
+import { useTabListState } from '@react-stately/tabs';
+import { CollectionChildren } from '@react-types/shared';
+import { forwardRef, HTMLAttributes, Key, useEffect, useState } from 'react';
+
+import { useDOMRef } from '../utils/dom';
+import { Tab } from './Tab';
+import { TabPanel } from './TabPanel';
+import { TabList, TabListWrapper, TabSelection } from './Tabs.style';
+
+type Props = {
+  defaultSelectedKey?: Key;
+  selectedKey?: Key;
+  onSelectionChange?: (index: Key) => void;
+  disabledKeys?: Key[];
+};
+
+type NativeAttrs = Omit<HTMLAttributes<unknown>, keyof Props>;
+
+type TabsProps = Props & NativeAttrs;
+
+const Tabs = forwardRef<HTMLDivElement, TabsProps>(
+  ({ children, ...props }, ref): JSX.Element => {
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    const ariaProps = { children: children as CollectionChildren<object>, ...props };
+    const state = useTabListState(ariaProps);
+    const tabsListRef = useDOMRef<HTMLDivElement>(ref);
+    const { tabListProps } = useTabList(ariaProps, state, tabsListRef);
+
+    const [activeTabStyle, setActiveTabStyle] = useState({
```

Comment:
Could these be moved into the style file? If you set a boolean here then you can add it as a prop to `TabSelection` rather than having css properties declared in the component.